src/test/java/no/unit/nva/datacite/DataCiteMdsCreateDoiHandlerTest.java

```diff
@@ -57,410 +72,336 @@
-        when(mockStatusLine.getStatusCode()).thenReturn(Response.Status.CREATED.getStatusCode());
-        when(mockCloseableHttpResponse.getStatusLine()).thenReturn(mockStatusLine);
-        when(mockDataCiteMdsConnection.postMetadata(any(), any())).thenReturn(mockCloseableHttpResponse);
-
-        when(mockDataCiteMdsConnection.postDoi(any(), any())).thenThrow(new IOException(MOCK_IO_EXCEPTION_MESSAGE));
-        when(mockDataCiteMdsConnection.deleteMetadata(any())).thenThrow(new IOException(MOCK_IO_EXCEPTION_MESSAGE));
-
-        DataCiteMdsCreateDoiHandler mockDataCiteMdsCreateDoiHandler =
-                new DataCiteMdsCreateDoiHandler(mockDataCiteMdsConnection, mockSecretCache);
-
-        GatewayResponse gatewayResponse = mockDataCiteMdsCreateDoiHandler.handleRequest(requestEvent, null);
-
-        GatewayResponse expectedResponse = new GatewayResponse();
-        expectedResponse.setStatusCode(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
-        expectedResponse.setErrorBody(ERROR_SETTING_DOI_URL_COULD_NOT_DELETE_METADATA);
-
-        assertEquals(expectedResponse.getStatusCode(), gatewayResponse.getStatusCode());
-        assertEquals(expectedResponse.getBody(), gatewayResponse.getBody());
+        HttpResponse<String> httpResponsePostMetadata = createHttpResponse(postMetadataResponseStream, SC_CREATED);
+        when(dataCiteMdsConnection.postMetadata(any(), any())).thenReturn(httpResponsePostMetadata);
+
+        when(dataCiteMdsConnection.postDoi(any(), any())).thenThrow(new IOException(""));
+        when(dataCiteMdsConnection.deleteMetadata(any())).thenThrow(new IOException(""));
+
+        InputStream input = requestWithHeaders(MOCK_URL, MOCK_KNOWN_INSTITUTION_ID, MOCK_DATACITE_XML);
+
+        dataCiteMdsCreateDoiHandler = new DataCiteMdsCreateDoiHandler(environment, dataCiteMdsConnection, secretCache);
+
+        dataCiteMdsCreateDoiHandler.handleRequest(input, output, context);
+        GatewayResponse gatewayResponse = objectMapper.readValue(output.toString(), GatewayResponse.class);
+        Problem problem = (Problem) gatewayResponse.getBodyObject(Problem.class);
+
+        assertThat(problem.getStatus(), is(Status.INTERNAL_SERVER_ERROR));
+        assertThat(problem.getTitle(), containsString(Status.INTERNAL_SERVER_ERROR.getReasonPhrase()));
+        assertThat(problem.getDetail(), containsString(ERROR_SETTING_DOI_URL_COULD_NOT_DELETE_METADATA));
+    }
+
+    private HttpResponse<String> createHttpResponse(InputStream inputStream, int httpStatus) throws IOException {
+
+        HttpResponse<String> httpResponse = mock(HttpResponse.class);
+        if (inputStream != null) {
+            String body = IoUtils.streamToString(inputStream);
+            when(httpResponse.body()).thenReturn(body);
+        }
+        when(httpResponse.statusCode()).thenReturn(httpStatus);
+
+        return httpResponse;
+    }
+
+    private InputStream jsonNodeToInputStream(JsonNode request) throws JsonProcessingException {
```

Comment:
You can use the HandlerUtils in nva-testutils to create requests and headers are available in TestHeaders in nva-testutils